framework/src/multiapps/MultiApp.C

```diff
@@ -439,9 +435,9 @@ MultiApp::restore(bool force)
   if (force || needsRestoration())
   {
     // Must be restarting / recovering so hold off on restoring
-    // Instead - the restore will happen in createApp()
+    // Instead - the restore will happen in initialSetup()
     // Note that _backups was already populated by dataLoad()
-    if (_apps.empty())
+    if (_fe_problem.getCurrentExecuteOnFlag() == EXEC_INITIAL)
```

Comment:
I was missing this line previously. This should pass framework normal and recover tests. We will need to fix stochastic_tools module because there are a few multiapps relying on the old behavior and a control for passing commandline options to subapp but late after subapp has been created. Will need @aeslaughter or @zachmprince 's help for fixing it.

Reply:
This doesn't make logical sense.  The original check had nothing to do with EXEC_INITIAL...

Either this check needs to be removed... or this is wrong.

Reply:
The original test is simply to check if this function called in `dataRestore` is from main-app's `initialSetup`. It can check if `_apps` is empty because at that point, MultiApp's `initialSetup` has not been called. I have to change this check because now `_apps` is constructed and this check is wrong. This line checking if the restore is happening in main-app's `initialSetup` will work in the old way as well.

Reply:
I'v double checked my claim above `This line checking if the restore is happening in main-app's initialSetup will work in the old way as well`. I will separate this line out in an independent commit.